### ISSUE TITLE
log_query config setting is not respected by query builder

### ISSUE THREAD

**Issue by rullzer** [nextcloud/server#1676]
When looking trough some requests I noticed I was missing some queries in the query log.
It turns our that all queries that are handled by the query builder (the way we want it) are not logged.

CC: @nickvergessen @MorrisJobke

**Comment by nickvergessen** [comment]
Tried adding:
```php
		if (\OC::$server->getConfig()->getSystemValue('log_query', false)) {
			\OC::$server->getLogger()->debug('DB QueryBuilder: {query}', [
				'query' => $this->getSQL(),
				'app' => 'core',
			]);
		}
```
in the `execute()` method of the `QueryBuilder`.

But then I receive: 
```
Technical details

    Remote Address: ::1
    Request ID: MZ1KfEVr9B6iqqNO72SK
    Type: Exception
    Code: 0
    Message: Session has been closed - no further changes to the session are allowed
    File: /home/nickv/Nextcloud/11/server/lib/private/Session/Memory.php
    Line: 120


Trace

#0 /home/nickv/Nextcloud/11/server/lib/private/Session/Memory.php(53): OC\Session\Memory->validateSession()
#1 /home/nickv/Nextcloud/11/server/lib/private/Security/CSRF/TokenStorage/SessionStorage.php(64): OC\Session\Memory->set('requesttoken', 'LKmKlT9hYX52PYt...')
#2 /home/nickv/Nextcloud/11/server/lib/private/Security/CSRF/CsrfTokenManager.php(58): OC\Security\CSRF\TokenStorage\SessionStorage->setToken('LKmKlT9hYX52PYt...')
#3 /home/nickv/Nextcloud/11/server/lib/public/Util.php(507): OC\Security\CSRF\CsrfTokenManager->getToken()
#4 /home/nickv/Nextcloud/11/server/lib/private/AppFramework/Middleware/Security/SecurityMiddleware.php(148): OCP\Util::callRegister()
#5 /home/nickv/Nextcloud/11/server/lib/private/AppFramework/Middleware/MiddlewareDispatcher.php(94): OC\AppFramework\Middleware\Security\SecurityMiddleware->beforeController(Object(OCA\Files\Controller\ViewController), 'index')
#6 /home/nickv/Nextcloud/11/server/lib/private/AppFramework/Http/Dispatcher.php(89): OC\AppFramework\Middleware\MiddlewareDispatcher->beforeController(Object(OCA\Files\Controller\ViewController), 'index')
#7 /home/nickv/Nextcloud/11/server/lib/private/AppFramework/App.php(114): OC\AppFramework\Http\Dispatcher->dispatch(Object(OCA\Files\Controller\ViewController), 'index')
#8 /home/nickv/Nextcloud/11/server/lib/private/AppFramework/Routing/RouteActionHandler.php(47): OC\AppFramework\App::main('ViewController', 'index', Object(OC\AppFramework\DependencyInjection\DIContainer), Array)
#9 [internal function]: OC\AppFramework\Routing\RouteActionHandler->__invoke(Array)
#10 /home/nickv/Nextcloud/11/server/lib/private/Route/Router.php(298): call_user_func(Object(OC\AppFramework\Routing\RouteActionHandler), Array)
#11 /home/nickv/Nextcloud/11/server/lib/base.php(994): OC\Route\Router->match('/apps/files/')
#12 /home/nickv/Nextcloud/11/server/index.php(40): OC::handleRequest()
#13 {main}
```

Not sure what goes wrong.

**Comment by nickvergessen** [comment]
Okay the problem is, that now the request ID is grabbed too early:
https://github.com/nextcloud/server/blob/851769adc821f0a8462081fe23afb5a4339eb7ad/lib/private/Server.php#L701-L701

Therefor the `CsrfTokenManager` is initialised with the first session, before base.php sets it in:
https://github.com/nextcloud/server/blob/9a2511fe9bd5716c49bc8a28cbea690f9bb0fe97/lib/base.php#L423-L423

Therefor writing the CSRF token into the session fails, because that still uses the old session which by that time is now closed and can not be written to anymore.

Basically DI nightmare. cc @LukasReschke
